a3fe/run/simulation.py

```diff
@@ -585,13 +550,41 @@ def update_paths(self, old_sub_path: str, new_sub_path: str) -> None:
             "crdfile": "somd.rst7",
         }
         for option, name in input_paths.items():
-            _write_simfile_option(
-                self.simfile_path, option, _os.path.join(self.input_dir, name)
-            )
+            setattr(self.engine_config, option, _os.path.join(self.input_dir, name))
+            
+        # Write updated config to file
+        self.engine_config.write_somd_config(self.output_dir, self.lam)
 
     def set_simfile_option(self, option: str, value: str) -> None:
```

Comment:
We can remove all set_simfile_options - the single source of truth should be the engine config, which is modified when you want to change settings (and this is always written out before running)